shepherd_bms/src/main.cpp

```diff
@@ -18,94 +19,86 @@ bool dischargeConfig[NUM_CHIPS][NUM_CELLS_PER_CHIP] = {};
 ChipData_t *testData;
 Timer mainTimer;
 ComputeInterface compute;
+WDT_T4<WDT1> wdt;
+
+Timer chargeTimeout;
+Timer boostTimer;
+Timer boostRechargeTimer;
+
+AccumulatorData_t *prevAccData = nullptr;
 
 uint32_t bmsFault = FAULTS_CLEAR;
 
-void testSegments()
+uint16_t overVoltCount = 0;
+uint16_t underVoltCount = 0;
+uint16_t overCurrCount = 0;
+uint16_t chargeOverVolt = 0;
+uint16_t overChgCurrCount = 0;
+
+enum
 {
-	currTime = millis();
-	if (lastVoltTemp + 1000 < currTime) {
-		lastVoltTemp = currTime;
-		//Handle Segment data collection test logic
-		testData = new ChipData_t[NUM_CHIPS];
+	BOOST_STANDBY,
+	BOOSTING,
+	BOOST_RECHARGE
+}BoostState;
 
-		//Retrieve ALL segment data (therms and voltage)
-		segment.retrieveSegmentData(testData);
 
-		for (int chip = 0; chip < NUM_CHIPS; chip++)
```

Comment:
can we move this for loop into its own function elsewhere (segment.cpp?) as discussed a main file that just lists methods and isnt cluttered with other stuff (even as simple as a printing for loop) would be most readable

Reply:
not going to add comment for all instances but same applies for many other iterating loops in this file

Reply:
I agree. I bet Nick wants to setup DEBUG macros, but I would kinda prefer making a print info function

Reply:
Sorry, debug for the Serial print loop. Others ofc would move elsewhere